lib/web_of_science/base_rest_retriever.rb

```diff
@@ -45,13 +37,13 @@ def next_batch
     # this is the maximum number that can be returned in a single query by WoS
     MAX_RECORDS = 100
 
-    attr_reader :batch_size, :query, :query_id, :path, :params
+    attr_reader :batch_size, :query, :path, :params
 
     delegate :client, to: WebOfScience
 
     # @return [Boolean] all records retrieved?
     def records_retrieved?
-      @batch_one.nil? ? false : records_retrieved == records_found
+      @batch_one.blank? ? false : records_retrieved == records_found
     end
```

Comment:
Switching the `nil?` to `blank?` since our new method deals with arrays, and we want the same logic that applies here to work for an empty array too